### PR TITLE
Refactor db access audit logger for better reuse

### PR DESCRIPTION
**Pull request by r0mant**
Another small refactoring PR which will allow for better reuse of the audit events emitting logic for MySQL integration.

### CODE REVIEW

File: lib/srv/db/common/audit.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+			Error:       trace.Unwrap(err).Error(),
+			UserMessage: err.Error(),
+		}
+	}
+	return a.cfg.StreamWriter.EmitAuditEvent(ctx, event)
+}
+
+// OnSessionEnd emits an audit event when database session ends.
+func (a *Audit) OnSessionEnd(ctx context.Context, session Session) error {
+	return a.cfg.StreamWriter.EmitAuditEvent(ctx, &events.DatabaseSessionEnd{
```
Comment (russjones) [inline lib/srv/db/common/audit.go:100]: Missing `trace.Wrap`.

**Review by russjones — APPROVED** [review]

File: lib/srv/db/common/audit.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+		},
+		DatabaseMetadata: events.DatabaseMetadata{
+			DatabaseService:  session.Server.GetName(),
+			DatabaseProtocol: session.Server.GetProtocol(),
+			DatabaseURI:      session.Server.GetURI(),
+			DatabaseName:     session.DatabaseName,
+			DatabaseUser:     session.DatabaseUser,
+		},
+	}
+	if err != nil {
```
Comment (a-palchikov) [inline lib/srv/db/common/audit.go:87]: which err?

File: lib/srv/db/common/audit.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+		cfg: config,
+	}, nil
+}
+
+// OnSessionStart emits an audit event when database session starts.
+func (a *Audit) OnSessionStart(ctx context.Context, session Session, err error) error {
+	event := &events.DatabaseSessionStart{
+		Metadata: events.Metadata{
+			Type: libevents.DatabaseSessionStartEvent,
+			Code: libevents.DatabaseSessionStartCode,
```
Comment (a-palchikov) [inline lib/srv/db/common/audit.go:64]: I added the cluster name to the event metadata [not so long ago](https://github.com/gravitational/teleport/pull/4994/) - something to consider here and below as well?

**Review by a-palchikov — APPROVED** [review]

File: lib/srv/db/common/audit.go (hunk 3 above)
Comment (r0mant) [reply lib/srv/db/common/audit.go]: Good point, added.

**Pull request merged**